process/heartbeat/interceptedHeartbeat_test.go

```diff
@@ -0,0 +1,190 @@
+		HardforkMessage: "hardfork message",
+	}
+	marshalizer := mock.MarshalizerMock{}
+	payloadBytes, err := marshalizer.Marshal(payload)
+	if err != nil {
+		return nil
+	}
+
+	return &heartbeat.HeartbeatV2{
+		Payload:         payloadBytes,
+		VersionNumber:   "version number",
+		NodeDisplayName: "node display name",
+		Identity:        "identity",
+		Nonce:           123,
+		PeerSubType:     uint32(core.RegularPeer),
+	}
+}
+
+func createMockInterceptedHeartbeatArg(interceptedData *heartbeat.HeartbeatV2) ArgInterceptedHeartbeat {
+	arg := ArgInterceptedHeartbeat{}
+	arg.Marshalizer = &mock.MarshalizerMock{}
+	arg.DataBuff, _ = arg.Marshalizer.Marshal(interceptedData)
+	arg.PeerId = "pid"
+
+	return arg
+}
+
+func TestNewInterceptedHeartbeat(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil data buff should error", func(t *testing.T) {
+		t.Parallel()
+
+		arg := createMockInterceptedHeartbeatArg(createDefaultInterceptedHeartbeat())
+		arg.DataBuff = nil
+
+		ihb, err := NewInterceptedHeartbeat(arg)
+		assert.Nil(t, ihb)
+		assert.Equal(t, process.ErrNilBuffer, err)
+	})
+	t.Run("nil marshalizer should error", func(t *testing.T) {
+		t.Parallel()
+
+		arg := createMockInterceptedHeartbeatArg(createDefaultInterceptedHeartbeat())
+		arg.Marshalizer = nil
+
+		ihb, err := NewInterceptedHeartbeat(arg)
+		assert.Nil(t, ihb)
+		assert.Equal(t, process.ErrNilMarshalizer, err)
+	})
+	t.Run("empty pid should error", func(t *testing.T) {
+		t.Parallel()
+
+		arg := createMockInterceptedHeartbeatArg(createDefaultInterceptedHeartbeat())
+		arg.PeerId = ""
+
+		ihb, err := NewInterceptedHeartbeat(arg)
+		assert.Nil(t, ihb)
+		assert.Equal(t, process.ErrEmptyPeerID, err)
+	})
+	t.Run("unmarshal returns error", func(t *testing.T) {
+		t.Parallel()
+
+		arg := createMockInterceptedHeartbeatArg(createDefaultInterceptedHeartbeat())
+		arg.Marshalizer = &mock.MarshalizerStub{
+			UnmarshalCalled: func(obj interface{}, buff []byte) error {
+				return expectedErr
+			},
+		}
+
+		ihb, err := NewInterceptedHeartbeat(arg)
+		assert.Nil(t, ihb)
+		assert.Equal(t, expectedErr, err)
+	})
+	t.Run("unmarshalable payload returns error", func(t *testing.T) {
+		t.Parallel()
+
+		interceptedData := createDefaultInterceptedHeartbeat()
+		interceptedData.Payload = []byte("invalid data")
+		arg := createMockInterceptedHeartbeatArg(interceptedData)
+
+		ihb, err := NewInterceptedHeartbeat(arg)
+		assert.Nil(t, ihb)
+		assert.NotNil(t, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		arg := createMockInterceptedHeartbeatArg(createDefaultInterceptedHeartbeat())
+
+		ihb, err := NewInterceptedHeartbeat(arg)
+		assert.False(t, ihb.IsInterfaceNil())
+		assert.Nil(t, err)
+	})
+}
+
+func Test_interceptedHeartbeat_CheckValidity(t *testing.T) {
+	t.Parallel()
+	t.Run("payloadProperty too short", testInterceptedHeartbeatPropertyLen(payloadProperty, false))
+	t.Run("payloadProperty too short", testInterceptedHeartbeatPropertyLen(payloadProperty, true))
```

Comment:
```suggestion
	t.Run("payloadProperty too long", testInterceptedHeartbeatPropertyLen(payloadProperty, true))
```
same for all tests description